### PR TITLE
Add acceptance tests for driver package

### PR DESCRIPTION
**Pull request by xosmig**

### CODE REVIEW

File: driver/testing/datastore_acc_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"testing"
+)
+
+func TestDatastoreAcc(t *testing.T) {
+	initDriverAcceptanceTest(t)
+
+	d := NewTestDriver(t)
+	ds, err := d.FindDatastore(TestDatastore)
```
Comment (mkuzmin) [inline driver/testing/datastore_acc_test.go]: test would become more readable if test data is just hardcoded inline

File: driver/testing/datastore_acc_test.go [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package testing
```
Comment (mkuzmin) [inline driver/testing/datastore_acc_test.go]: if we split Packer tests from driver tests, a separate package is not required

File: driver/testing/vm_acc_test.go [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+	d = NewTestDriver(t)
+
+	template, err := d.FindVM(TestTemplate) // Don't destroy this VM!
+	if err != nil {
+		t.Fatalf("Cannot find template vm '%v': %v", TestTemplate, err)
+	}
+
+	log.Printf("[DEBUG] Clonning VM")
+	vmName = NewVMName()
+	vm, err = template.Clone(&driver.CloneConfig{
```
Comment (mkuzmin) [inline driver/testing/vm_acc_test.go]: linked clones are faster, but can affect some tests (like snapshots)

File: driver/testing/vm_acc_test.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	}
+
+	vmDestructor = func() {
+		log.Printf("[DEBUG] Removing the clone")
+		if err := vm.Destroy(); err != nil {
+			t.Errorf("!!! ERROR REMOVING VM '%v': %v!!!", vmName, err)
+		}
+
+		// Check that the clone is no longer exists
+		if _, err := d.FindVM(vmName); err == nil {
```
Comment (mkuzmin) [inline driver/testing/vm_acc_test.go]: `Destroy()` is synchronous. I suppose this case is impossible.

**Comment by xosmig** [comment]
@mkuzmin fixed the issues. Could you review it again, please?

**Pull request merged**